lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1634,6 +1634,19 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
   };
 
   switch (II->getIntrinsicID()) {
+  case Intrinsic::assume: {
+    // llvm.assume translation is currently supported only within
+    // SPV_INTEL_optimization_hints extension, ignore it otherwise, since it's
+    // an optimization hint
+    if (BM->isAllowedToUseExtension(
+          ExtensionID::SPV_INTEL_optimization_hints)) {
+      BM->addCapability(CapabilityOptimizationHintsINTEL);
```

Comment:
No need to manually insert capability, just register it here: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/10cdc73aca7ece07d7428676a84f6e15a06a7c58/lib/SPIRV/libSPIRV/SPIRVEnum.h#L313-L315